crates/core/src/action.rs

```diff
@@ -216,14 +216,15 @@ pub enum Action {
     ReturnInitializationResult(Result<Initialization, String>),
 
     /// Gets dispatched when a zome function call starts.
-    /// There is no reducer for this action so this does not change state
-    /// (hence "Signal").
-    /// Is received as signal in the nodejs waiter to attach wait conditions.
     QueueZomeFunctionCall(ZomeFnCall),
 
     /// return the result of a zome WASM function call
     ReturnZomeFunctionResult(ExecuteZomeFnResponse),
 
+    InvokeHdkFunction((ZomeFnCall, HdkFnCall)),
+
+    ReturnHdkFunction((ZomeFnCall, HdkFnCall, HdkFnCallResult)),
```

Comment:
```suggestion
    TrackZomeFnCallEnd((ZomeFnCall, HdkFnCall, HdkFnCallResult)),
```
Or something like that.

Also `HdkFnCall` is confusing because what's the difference between it and `ZomeFnCall`

Maybe `TrackZomeFnCallData`  & `TracZomeFnCallResult` ?